### DISCUSSION TITLE
My project requires an admin panel, so I've decided to implement an admin site generator.

### DISCUSSION THREAD

**Discussion by y3l1n4ung** [serverpod/serverpod#1324] [Ideas]
My project requires an admin panel, so I've decided to implement an admin site generator. However, I'm unsure if I'm implementing it correctly. Here is my current implementation...
```dart
#admin_endpoints.dart
@AdminSiteRegister(models: [
  Movie,
  // Genres,
  // MovieLink,
  // SubtitleLink,

  // Series,
  // Season,
], requiredLogin: false, isAdminOnly: false)
class MovieModelAdmin extends ModelAdmin {
  /// final Set<Scope> requiredScope = {Scope.admin};
  final List<String> searchField = [Movie.t.title.columnName];
}```

```dart
#admin_annotation.dart
abstract class BaseEndpoint<S, T> {
  Future<void> create(S s, T row);
  Future<T?> read(S s, int id);
  Future<bool> update(S s, T row);
  Future<int> delete(S s, int id);
  Future<List<T>> getAll(S s, {int limit = 20, int currentPage = 1});
}


```
after running `dart run build_runner build -d  ` it will generate something like that
```dart
#admin_endpoint.admin.dart
 ```
![Screenshot 2023-09-02 at 5 54 40 PM](https://github.com/serverpod/serverpod/assets/40236753/2ac418d6-3890-4b31-88ff-ef3fd93c7ab1)

This is my current progress. I also want to implement 'some_cli create adminuser' to create a superuser for the first time and automate build a user interface for the admin panel.

I appreciate and welcome any feedback or suggestions.

**Comment by vlidholt** [comment]
Cool, we have been talking about add a feature which looks similar to this. It would be pretty cool to be able to automatically generated add public crud operations to tables too.
